yt/data_objects/static_output.py

```diff
@@ -817,6 +818,10 @@ def _get_field_info(self, ftype, fname=None):
         # storing this condition before altering it
         guessing_type = ftype == "unknown"
         if guessing_type:
+            if fname in self.field_info._ambiguous_field_names:
```

Comment:
It may be more robust to create a method for this, to prevent accessing an "hidden" attribute (i.e. starting with a `_`).
```suggestion
            if self.field_info.is_ambiguous_field_name(fname):
```